docs/project_structue.md

```diff
@@ -0,0 +1,63 @@
+# Project Structure
+- configs: Satellite configs.
+- internal: Core, Api, and common utils.
+- internal/pkg: Sharing with Core and Plugins, such as api and utils.
+- internal/satellite: The core of Satellite.
+- plugins: Contains all plugins.
+- plugins/{type}: Contains the plugins of this {type}. Satellite has 6 plugin types, which are collector, queue, parser, filter, client, and forward.
+- plugins/{type}/define{type}: Contains the plugin define.
```

Comment:
move  API defines to plugins dir

Reply:
Why need duplicate `{type}`? And define is a verb, usually, a package name is a noun.

Reply:
just to avoid package conflicts, let me try again.

Reply:
What conflict?

Reply:
already fixed

Reply:
> What conflict?

It's my mistake

Reply:
could I merge it? According to chao's advice, the plugin was decoupled.